src/ContentMessages.js

```diff
@@ -52,7 +52,7 @@ const MAX_HEIGHT = 600;
  *  and a thumbnail key.
  */
 function createThumbnail(element, inputWidth, inputHeight, mimeType) {
-    const deferred = q.defer();
+    const deferred = Promise.defer();
```

Comment:
Yup, we should aim to switch over, but I think doing it right will be quite a lot of work, so I think we should leave it for now and gradually convert them in future. In some cases it will be quite fiddly.